lib/moon/design/form/field/error.ex

```diff
@@ -35,8 +37,11 @@ defmodule Moon.Design.Form.Field.Error do
       <#slot>
         <Icon name="generic_info" :if={@has_error_icon} />
       </#slot>
-      <Surface.Components.Form.ErrorTag />
+      <Moon.Design.Form.Field.ErrorTag form={get_changeset(@form)} />
```

Comment:
- function name is misleading because it's not necessarily returning a changeset
- `to_form` helper that generates the struct `%Phoenix.HTML.Form{}` accepts maps as arguments in addition to changesets. In that case, `source` will be a plain map.
  - If `source` is a map, and doesn't have `:errors` key, `Moon.Design.Form.Field.ErrorTag` component will crash

This implementation works only for a subset of what default LiveView offers.

Reply:
```suggestion
      <Moon.Design.Form.Field.ErrorTag form={@form[field_name])} />
```

With the `Access` behaviour you can extract the `Phoenix.HTML.FormField` which will also contain the `:errors` key

Reply:
not, it will not worl. unfortunatly `@form[field_name].errors` is empty until apply. the same as `@form.errors`.

Reply:
What do you mean by until apply?

What you are saying sounds like a problem upstream, and not in the component

You can check the behaviour I just mentioned fully functioning on [Chris's TodoTrek demo](https://github.com/chrismccord/todo_trek/blob/4d7357428993ce46625e1d39cb0fda9c0a832fc6/lib/todo_trek_web/components/core_components.ex#L277)